src/main/webapp/js/gensim.js

```diff
@@ -16,15 +17,50 @@ async function postGensim(url, data, query, callback=handleGensimResponse) {
             headers: {
                 'Content-Type': 'application/json'
             },
-            body: JSON.stringify({'query': query,
-                                'ytCaptions': data})
-        });
+            body: JSON.stringify({'data': data,
+                                'v_id': v_id})
+        }).catch(error => { throw error });
 
         callback(response);
-        // return response;
+
+    } catch(err) {
+        console.log(err);
+        $('#query-output table').empty();
```

Comment:
Line 28 - 31 seems repeated, could we create a function handleError to do the UI change?